controllers/module_reconciler.go

```diff
@@ -264,33 +259,20 @@ func (r *ModuleReconciler) getNodesListBySelector(ctx context.Context, mod *kmmv
 func (r *ModuleReconciler) handleBuild(ctx context.Context,
 	mod *kmmv1beta1.Module,
 	km *kmmv1beta1.KernelMapping,
-	kernelVersion string,
-	containerImage string) (bool, error) {
+	kernelVersion string) (bool, error) {
 
-	//if theres no build specified skip this section
-	if !r.willBuild(mod, km) {
-		return false, nil
-	}
-	//if build is specified AND sign is specified then we want to build an intermediate image
-	// and let sign produce the one specified in containerImage
-	if r.willSign(mod, km) {
-		containerImage = r.appendToTag(containerImage, mod.Namespace+"_"+mod.Name+"_kmm_unsigned")
-	}
-
-	// build is specified and containerImage is either the final image, or the intermediate image tag depending on if sign is defined
-	// either way if containerImage exists we can skip building it
-	exists, err := r.checkImageExists(ctx, mod, km, containerImage)
+	shouldSync, err := r.buildAPI.ShouldSync(ctx, *mod, *km)
 	if err != nil {
-		return false, fmt.Errorf("failed to check existence of image %s for kernel %s: %w", containerImage, kernelVersion, err)
+		return false, fmt.Errorf("could not check if build synchronization is needed: %w", err)
 	}
-	if exists {
+	if !shouldSync {
 		return false, nil
 	}
 
-	logger := log.FromContext(ctx).WithValues("kernel version", kernelVersion, "image", containerImage)
+	logger := log.FromContext(ctx).WithValues("kernel version", kernelVersion, "image", km.ContainerImage)
```

Comment:
Not sure how that would play out if the `containerImage` is defined at the top-level, i.e. if `km.ContainerImage` is empty. Probably outside of the scope of this PR because I don't think it introduces a bug, but the code could use some consolidation there.

Reply:
This is a valid point and I think that we have ourselves a bug here, although we have to test and verify it. The bug being: the `ModuleLoaderContainer.Spec.ContainerImage` field is never used, we only use the `ModuleLoaderContainerSpec.KernelMappings[].ContainerImage` field instead. 

I suggest we open an issue for this bug, verify it and resolve it in a separate PR. WDYT?